pkg/controller/scyllaoperatorconfig/sync_clusterdomain.go

```diff
@@ -0,0 +1,31 @@
+package scyllaoperatorconfig
+
+import (
+	"context"
+	"fmt"
+
+	osscyllav1alpha1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1alpha1"
+	"github.com/scylladb/scylla-operator/pkg/pointer"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (socc *Controller) syncClusterDomain(
+	ctx context.Context,
+	soc *osscyllav1alpha1.ScyllaOperatorConfig,
+	status *osscyllav1alpha1.ScyllaOperatorConfigStatus,
+) ([]metav1.Condition, error) {
+	var progressingConditions []metav1.Condition
+
+	if soc.Spec.ConfiguredClusterDomain != nil && len(*soc.Spec.ConfiguredClusterDomain) != 0 {
```

Comment:
technically (given it's forbidden in validation), if it's not nil we should use it and if it's empty value it should error out (instead of silently autodiscovering it when specified)